### PR TITLE
BBE: Add input to accept search terms

### PR DESCRIPTION
**Pull request by aneeshd16**
Fixes Automattic/martech#1556

## Proposed Changes

* Adds a new text field to the site options step in the BBE flow.
image

* On the content collection form, modified the copy for the Home page:
image
 

## Testing Instructions

* Apply D102526-code.
* Go to `/start/do-it-for-me`.
* Select any of the options on the New/Existing site screen.
* On the site options screen, confirm that you can see the new text field for accepting search terms.
* Continue through the flow and complete the purchase.
* On the content collection form, confirm that the copy for the Home page matches the one in the screenshot above.
* Additional testing instructions in D102526-code.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)? (TODO)
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/signup/steps/site-options/site-options.tsx [hunk 1]
```diff
@@ -104,6 +111,26 @@ const SiteOptions: React.FC< Props > = ( {
...
+			{ acceptSearchTerms && (
+				<FormFieldset className="site-options__form-fieldset">
+					<FormLabel htmlFor="searchTerms" optional={ true }>
+						{ translate( 'Search terms' ) }
+					</FormLabel>
+					<FormInput
+						name="searchTerms"
+						id="searchTerms"
+						value={ formValues.searchTerms }
+						// isError={ Boolean( searchTermsError ) }
```
Comment (ddc22) [inline client/signup/steps/site-options/site-options.tsx]: Cleanup on line number 123 😁

**Review by ddc22 — APPROVED** [review]
LGTM minor code feedback below! 

Also a suggestion that you could probably pick up later. Maybe you can do something to prevent this line break, (Increase the width of the container and text boxes, make the font smaller).
| Existing | Suggestion |
|--|--|
|image|  image |

If you are going to do this change however, it shouldn't ideally effect the other flows I guess.
:shipit:

File: client/signup/steps/site-options/site-options.tsx (hunk 1 above)
Comment (aneeshd16) [reply client/signup/steps/site-options/site-options.tsx]: Haha!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7833362

Thank you @aneeshd16 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7833362).

**Pull request merged**